src/util/include/util/RAII_Wrappers.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Xournal++
+ *
+ * RAII wrappers for C library classes
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <cairo.h>
+#include <gtk/gtk.h>
+
+#include "CLibrariesSPtr.h"
+
+namespace xoj::util {
+
+inline namespace raii {
+namespace specialization {
+
+class CairoHandler {
+public:
+    constexpr static auto ref = cairo_reference;
+    constexpr static auto unref = cairo_destroy;
+};
+
+class CairoSurfaceHandler {
+public:
+    constexpr static auto ref = cairo_surface_reference;
+    constexpr static auto unref = cairo_surface_destroy;
+};
+
+template <class object_type>
+class GInitiallyUnownedHandler {
+    static_assert(std::is_base_of<GInitiallyUnowned, object_type>::value, "Class must inherit GInitiallyUnowned");
```

Comment:
I don't think this will work, because C inheritance is self implemented and cannot be determined by the compiler.